pkg/metrics/iterable_series.go

```diff
@@ -41,7 +42,9 @@ func NewIterableSeries(callback func(*Serie), chanSize int, bufferSize int) *Ite
 func (series *IterableSeries) Append(serie *Serie) {
 	series.callback(serie)
 	atomic.AddUint64(&series.count, 1)
-	series.ch.Put(serie)
+	if err := series.ch.Put(serie); err != nil {
+		log.Errorf("Error when addding a serie %v", err)
```

Comment:
If the channel is closed mid-flush, this will produce massive amount of identical log entries. It might be better to handle this higher in the calll stack and (cleanly?) abort the flush operation, since there's no way to make progress now.

Reply:
This error happens when the serializer has a fatal error during the serialization and drop all remaining series. I don't think we should care too much about this kind of error. I didn't handle the error initially because the serializer also report an error. 
There is no easy way to handle the error at an higher level as `Append` is called from different locations.
What I can do is to store the error and report once in `SenderStopped()` but it makes the assumption that `Put` always return the same error which is not great.
For these reason, I will tend to let the code as if. What do you think? 

Note: Ideally we should have a way to report errors with throttling.

Reply:
Don't agree, even if it is unlikely, repeated messages do not help. They will either fill the disk until the system runs out of space, or erase the really interesting infromation (what happened before the crash) if the logs are rotated based on size. 

It also doesn't seem possible to goroutine "crash" without bringing down the entire process with it. AIUI it is more likely for serializer to stop reading the channel without closing it due to a logic bug, in which case it won't necessarily produce any logs to identify the problem.

So I think we should log, but log only once. We could add a boolean to IterableSerie and skip calling `Put()` after it returns an error for the first time. WDYT?